src/effects/effectslot.cpp

```diff
@@ -161,11 +160,11 @@ void EffectSlot::loadEffect(EffectPointer pEffect) {
             addEffectButtonParameterSlot();
         }
 
-        foreach (EffectParameterSlotPointer pParameter, m_parameters) {
+        for (EffectParameterSlotPointer pParameter : m_parameters) {
             pParameter->loadEffect(pEffect);
         }
 
-        foreach (EffectButtonParameterSlotPointer pParameter, m_buttonParameters) {
+        for (EffectButtonParameterSlotPointer pParameter : m_buttonParameters) {
```

Comment:
> const in this context means that the QSharedPointer is const. The pointer carried by a shared pointer is always non const.

Thanks for clarifying. I was a bit confused how those loops were still working with const.